src/mongo/db/storage/mmap_v1/mmap_windows.cpp

```diff
@@ -185,7 +185,7 @@ void* MemoryMappedFile::map(const char* filenameIn, unsigned long long& length)
     filename[255] = 0;
     {
         size_t len = strlen(filename);
-        for (size_t i = len - 1; i >= 0; i--) {
+        for (int i = (len > 0 ? static_cast<int>(len - 1) : -1); i >= 0; i--) {
```

Comment:
I am only concerned about comparison between unsigned integer variable and 0. Even simpler it may be if len will have type int, because size of 'filename' is not more than 256.